recompile only outdated when container is selected

nittka opened the pull request:
As recompilation is not possible on an ily file anymore, it may be less easy to invoke recompilation on all including files (which may be spread over different folders), especially if compile on save is deactivated.
This pull request introduces a preference value controlling whether all files or only those marked as outdated will be processed, when recompilation is invoked on a folder or a project.

(My hope was that this makes #127 obsolete, as the Elysium's compiler preference page is completely different from the standard Xtext preference page and hence allowing project specific settings may not be easy. However, the use case there - compile on save in one project but not another - remains)

thSoft commented:
Yes, this behavior needs to be disambiguated. I think the user should make it explicit whether (s)he wants to recompile every file or just the outdated ones, so I suggest that a container should have two separate commands.

nittka commented:
So, rather than the preference, a "Recompile outdated" command should be added (and the other be renamed to "Recompile all")? Recompile viewed and recompile edited are not affected.
Note, that the selection may involve both multiple single files and multiple containers.

thSoft commented:
Yes, this would be great.

thSoft commented on org.elysium.parent/org.elysium.ui/src/org/elysium/ui/compiler/handlers/RecompileSelectedHandler.java:
```diff
@@ -25,14 +27,17 @@
 
 	@Override
 	public Object execute(ExecutionEvent event) throws ExecutionException {
+		compileOutdatedOnly = event.getCommand().getId().endsWith("outdatedOnly");
```
Please add a commandParameter to the command in the plugin.xml instead of deciding based on the command's name.

nittka replied:
I tried that. But I failed. Distinguishing the cases using command ids seemed a valid alternative. But I will check again, if I somehow can add a dummy parameter to the event.

nittka commented:
It seems, a command parameter would make sense only if there was only one recompile command and the user is asked every time whether all or only outdated files are to be compiled. Is that what you intended?

Otherwise two command IDs are necessary anyway and distinguishing the two cases by evaluating the ID seems a valid approach.

thSoft commented:
There can be one command definition (i.e. a command element under the org.eclipse.ui.commands extension point) which has a commandParameter, and two menu commands (i.e. command elements under the org.eclipse.ui.menus extension point) which can specify arguments (i.e. parameter elements) for the called commands.

nittka commented:
Thanks for the hint. I missed the possibility of providing the command parameter in the menu entry.
I added the command parameter, set the value in the compile-outdated menu entry, rebased on master and modified the original logic slightly. Rather than restricting the "outdated" logic to containers, now from all selected elements only outdated files are compiled.
I.e. the menu entry Recompile All compiles all ly files (derived from selected files and containers), and Recompile Outdated compiles only those which are marked as outdated.

If we wanted enablement of the entry to work perfectly (Recompile outdated is enabled only if there are outdated files) we would have to use a more elaborate logic, e.g. separate commands and handlers, because the isEnabled-method lacks the information which of the two menu entries is asking.

thSoft commented:
Thank you, this is sufficiently sophisticated for now.
